my-app/src/store/StocksStore/StocksStore.ts

```diff
@@ -0,0 +1,153 @@
+import { getStocksListParams, IStocksStore } from "@config/types";
+import ApiStore from "@store/ApiStore/ApiStore";
+import { CoinApi, CoinModels, normalizeCoin } from "@store/models/Coin";
+import {
+  CollectionModel,
+  getInitialCollectionModel,
+  linearizeCollection,
+  normalizeCollection,
+} from "@store/models/shared/collection";
+import rootStore from "@store/RootStore";
+import { queryFunction } from "@store/StocksStore/queryFunction.ts/queryFunction";
+import {
+  action,
+  computed,
+  IReactionDisposer,
+  makeObservable,
+  observable,
+  reaction,
+  runInAction,
+  toJS,
+} from "mobx";
+
+import { Meta } from "../meta";
+import {
+  normalizeStockItems,
+  StockItemsApi,
+  StockItemsModels,
+} from "../models";
+import { ILocalStore } from "../useLocalStore/useLocalStore";
+
+const BASE_URL = "https://api.coingecko.com/api/v3/coins/";
+
+type PrivateFields = "_list" | "_meta" | "_coin" | "_query";
+
+export default class StocksStore implements IStocksStore, ILocalStore {
+  private readonly _apiStore = new ApiStore(BASE_URL);
+  private _list: CollectionModel<string, StockItemsModels> =
+    getInitialCollectionModel();
+  private _coin: CoinModels | null = null;
+  private _meta: Meta = Meta.initial;
+  private _query: string = "";
+
+  constructor() {
+    makeObservable<StocksStore, PrivateFields>(this, {
+      _coin: observable.ref,
+      _list: observable.ref,
+      _meta: observable,
+      _query: observable,
+      query: computed,
+      list: computed,
+      meta: computed,
+      coin: computed,
+      getStocksList: action,
+      getCoin: action,
+      setQuery: action,
+    });
+  }
+
+  get list(): StockItemsModels[] {
+    return linearizeCollection(this._list);
+  }
+
+  get meta(): Meta {
+    return this._meta;
+  }
+
+  get coin(): CoinModels | null {
+    return this._coin;
+  }
+
+  get query(): string {
+    return this._query;
+  }
+
+  public setQuery = (getQuery: string): void => {
+    this._query = getQuery;
+    console.log(getQuery);
+    // this.getStocksList();
+  };
+
+  async getStocksList(params: getStocksListParams): Promise<void> {
+    this._meta = Meta.loading;
+    this._list = getInitialCollectionModel();
+
+    const query = queryFunction({
+      query: this._query !== "" ? this._query : "",
```

Comment:
а в этом есть смысл?.. мне кажется, тернарный оператор тут не нужен
```suggestion
      query: this._query,
```